listenbrainz_spark/stats/common/listening_activity.py

```diff
@@ -134,29 +145,34 @@ def get_time_range(stats_range: str) -> Tuple[datetime, datetime, relativedelta,
         # compute listening activity for each day, include weekday in date format
         step = relativedelta(days=+1)
         date_format = "%A %d %B %Y"
+        spark_date_format = "EEEE d MMMM y"
     elif stats_range == "month":
         from_offset = relativedelta(months=-2, day=1)  # start of the previous to previous month
         to_offset = relativedelta(months=+2)
         # compute listening activity for each day but no weekday
         step = relativedelta(days=+1)
         date_format = "%d %B %Y"
+        spark_date_format = "d MMMM y"
     elif stats_range == "quarter":
         from_offset = get_two_quarters_ago_offset(latest_listen_date)
         to_offset = relativedelta(months=+6)
         # compute listening activity for each week with date format as day
         step = relativedelta(weeks=+1)
         date_format = "%d %B %Y"
+        spark_date_format = "d MMMM y"
```

Comment:
Thinking again, this won't work for quarters. For quarters we want to round off each timestamp to week which is not simple. Need to rethink quarterly stats or special case the query for this.

Reply:
how does this comment impact this PR? Merge for now and fix later?

Reply:
As per IRC discussion, modified to use daily steps here instead of weekly.